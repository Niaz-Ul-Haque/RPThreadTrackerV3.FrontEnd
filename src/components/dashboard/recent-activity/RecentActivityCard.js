import React, { Component } from "react";
import PropTypes from 'prop-types';
import { Card, CardHeader, CardBlock } from 'reactstrap';
import RecentActivityRow from './RecentActivityRow';

const propTypes = {
};

const defaultProps = {
};

class RecentActivityCard extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	render() {
		return (
			<Card className="recent-activity-card">
				<CardHeader>
					<i className="fa fa-align-justify"></i> Recent Activity
				</CardHeader>
				<CardBlock className="card-body">
					<RecentActivityRow />
					<RecentActivityRow />
					<RecentActivityRow />
					<RecentActivityRow />
					<RecentActivityRow />
				</CardBlock>
			</Card>
		)
	}
}

RecentActivityCard.propTypes = propTypes;
RecentActivityCard.defaultProps = defaultProps;

export default RecentActivityCard;
