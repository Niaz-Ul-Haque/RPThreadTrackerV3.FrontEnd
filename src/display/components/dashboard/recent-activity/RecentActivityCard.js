import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardBlock } from 'reactstrap';
import RecentActivityRow from './RecentActivityRow';

const propTypes = {
	threads: PropTypes.arrayOf(PropTypes.shape({})).isRequired
};

const RecentActivityCard = (props) => {
	const { threads } = props;
	return (
		<Card className="recent-activity-card">
			<CardHeader>
				<i className="fa fa-align-justify" /> Recent Activity
			</CardHeader>
			<CardBlock className="card-body">
				{threads.map(thread => <RecentActivityRow thread={thread} key={thread.id} />)}
			</CardBlock>
		</Card>
	);
};
RecentActivityCard.propTypes = propTypes;
export default RecentActivityCard;
