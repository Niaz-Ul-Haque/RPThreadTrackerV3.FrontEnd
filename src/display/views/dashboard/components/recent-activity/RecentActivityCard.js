import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardHeader, CardBlock } from 'reactstrap';
import RecentActivityRow from './RecentActivityRow';

const propTypes = {
	threads: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
	archiveThread: PropTypes.func.isRequired,
	openUntrackThreadModal: PropTypes.func.isRequired,
	markThreadQueued: PropTypes.func.isRequired
};

const RecentActivityCard = (props) => {
	const { threads, archiveThread, openUntrackThreadModal, markThreadQueued } = props;
	return (
		<Card className="recent-activity-card">
			<CardHeader>
				<i className="fa fa-align-justify" /> Recent Activity
			</CardHeader>
			<CardBlock className="card-body">
				{threads.map(threadData =>
					(<RecentActivityRow
						threadData={threadData}
						key={threadData.thread.threadId}
						archiveThread={archiveThread}
						openUntrackThreadModal={openUntrackThreadModal}
						markThreadQueued={markThreadQueued}
					/>))}
			</CardBlock>
		</Card>
	);
};
RecentActivityCard.propTypes = propTypes;
export default RecentActivityCard;
