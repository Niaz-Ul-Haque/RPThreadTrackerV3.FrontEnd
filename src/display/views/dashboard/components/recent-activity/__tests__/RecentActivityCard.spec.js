// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../../../../../config/tests/helpers.unit';
import RecentActivityCard from '../RecentActivityCard';
// #endregion imports

// #region mocks
jest.mock('../RecentActivityRow', () => 'RecentActivityRow');
jest.mock('../../NoThreadsMessage', () => 'NoThreadsMessage');
jest.mock('../../NoCharactersMessage', () => 'NoCharactersMessage');
jest.mock('../../NoActiveCharactersMessage', () => 'NoActiveCharactersMessage');
jest.mock('../../NoRecentActivityMessage', () => 'NoRecentActivityMessage');
jest.mock('../../../../../shared/LoadingIndicator', () => 'LoadingIndicator');
// #endregion mocks

const createTestProps = propOverrides => ({
	allThreads: [],
	archiveThread: jest.fn(),
	characters: [],
	loadingInProgress: false,
	markThreadQueued: jest.fn(),
	openUntrackThreadModal: jest.fn(),
	recentActivityThreads: [],
	...propOverrides
});
const createTestPropsLoading = () => createTestProps({ loadingInProgress: true });
const createTestPropsNoCharacters = () => createTestProps({});
const createTestPropsNoActiveCharacters = () => createTestProps({
	characters: [{ isOnHiatus: true }, { isOnHiatus: true }]
});
const createTestPropsNoThreads = () => createTestProps({ characters: [{}, {}] });
const createTestPropsNoRecentActivity = () => createTestProps({
	characters: [{}, {}], allThreads: [{}, {}, {}]
});
const createTestPropsRecentActivity = () => createTestProps({
	characters: [{}, {}],
	allThreads: [{}, {}, {}],
	recentActivityThreads: [{ thread: { threadId: 1, userTitle: 'Recent 1' } }, { thread: { threadId: 2, userTitle: 'Recent 2' } }]
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot when threads are loading', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsLoading()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot when user has no characters', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsNoCharacters()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot when user has no active characters', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsNoActiveCharacters()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot when user has no threads', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsNoThreads()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot when user has no recent activity', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsNoRecentActivity()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot when user has recent activity', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsRecentActivity()} />);
			expect(element).toMatchSnapshot();
		});
	});
	describe('content', () => {
		it('should render item for each thread', () => {
			const element = shallow(<RecentActivityCard {...createTestPropsRecentActivity()} />);
			const rows = getSpecWrapper(element, 'recent-activity-card-row');
			expect(rows).toHaveLength(2);
		});
	});
});
