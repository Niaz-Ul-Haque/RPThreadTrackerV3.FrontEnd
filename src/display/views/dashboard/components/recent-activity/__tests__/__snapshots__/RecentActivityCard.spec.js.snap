// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot when threads are loading 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <LoadingIndicator
      style={
        Object {
          "height": 50,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": 50,
        }
      }
    />
  </CardBlock>
</Styled(Card)>
`;

exports[`rendering snapshots should render valid snapshot when user has no active characters 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <NoActiveCharactersMessage />
  </CardBlock>
</Styled(Card)>
`;

exports[`rendering snapshots should render valid snapshot when user has no characters 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <NoCharactersMessage />
  </CardBlock>
</Styled(Card)>
`;

exports[`rendering snapshots should render valid snapshot when user has no recent activity 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <NoRecentActivityMessage />
  </CardBlock>
</Styled(Card)>
`;

exports[`rendering snapshots should render valid snapshot when user has no threads 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <NoThreadsMessage />
  </CardBlock>
</Styled(Card)>
`;

exports[`rendering snapshots should render valid snapshot when user has recent activity 1`] = `
<Styled(Card)
  className="recent-activity-card"
>
  <CardHeader
    tag="div"
  >
    <i
      className="fas fa-bolt"
    />
     Recent Activity
  </CardHeader>
  <CardBlock
    className="card-body"
    tag="div"
  >
    <RecentActivityRow
      archiveThread={[MockFunction]}
      data-spec="recent-activity-card-row"
      key="1"
      markThreadQueued={[MockFunction]}
      openUntrackThreadModal={[MockFunction]}
      threadData={
        Object {
          "thread": Object {
            "threadId": 1,
            "userTitle": "Recent 1",
          },
        }
      }
    />
    <RecentActivityRow
      archiveThread={[MockFunction]}
      data-spec="recent-activity-card-row"
      key="2"
      markThreadQueued={[MockFunction]}
      openUntrackThreadModal={[MockFunction]}
      threadData={
        Object {
          "thread": Object {
            "threadId": 2,
            "userTitle": "Recent 2",
          },
        }
      }
    />
  </CardBlock>
</Styled(Card)>
`;
